### PR TITLE
[EG]Add healthcare API events

### PR DESCRIPTION
**Pull request by mssfang**
- Add healthcare events

### CODE REVIEW

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /** {@inheritDoc} */
+    @Override
+    public HealthcareFhirResourceCreatedEventData setResourceFhirAccount(String resourceFhirAccount) {
+        super.setResourceFhirAccount(resourceFhirAccount);
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public HealthcareFhirResourceCreatedEventData setResourceFhirId(String resourceFhirId) {
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java:28]: Why do we have setter methods on a type that is annotated with `@Immutable`?

Comment (mssfang) [reply sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java:28]: change to annotation Fluent

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java [hunk 2]
```diff
@@ -4,10 +4,10 @@
 
 package com.azure.messaging.eventgrid.systemevents;
 
-import com.azure.core.annotation.Immutable;
+import com.azure.core.annotation.Fluent;
 
 /** Schema of the Data property of an EventGridEvent for a Microsoft.HealthcareApis.FhirResourceCreated event. */
-@Immutable
+@Fluent
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java:10]: Was this manually changed? We should change this through customization.

Comment (mssfang) [reply sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java:11]: updated with cusotmizaton now

File: sdk/eventgrid/azure-messaging-eventgrid/CHANGELOG.md [hunk 3]
```diff
@@ -3,13 +3,9 @@
 ## 4.11.0-beta.1 (Unreleased)
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/CHANGELOG.md:3]: Update the version and release date too.

**Review by srnagar — APPROVED** [review]

**Pull request merged**